src/server.c

```diff
@@ -2127,6 +2127,54 @@ void loadingCron() {
     }
 }
 
+void blockingOperationStarts() {
+    updateCachedTime(0);
+    server.blocked_last_cron = server.mstime;
+}
+
+void blockingOperationEnds() {
+    server.blocked_last_cron = 0;
+}
+
+/* Do some of the tasks of the serverCron that are needed while we're blocked. */
+void whileBlockedCron() {
+    /* Here we may want to perform some cron jobs (normally done server.hz times
+     * per second). */
+
+    /* Since this function depends on a call to blockingOperationStarts, let's
+     * make sure it was done. */
+    serverAssert(server.blocked_last_cron);
+
+    /* Current only active defrag is performed here, and since it's not enabled
+     * by default. For now, just exit immediately to reduce any minor overheads
+     * of the code below. */
+    if (!server.active_defrag_enabled)
+        return;
+
+    mstime_t latency;
+    latencyStartMonitor(latency);
+
+    /* In some cases we may be called with big intervals, so we may need to do
+     * extra work here. This is because some of the functions in serverCron rely
+     * on the fact that it is performed every 10 ms or so. For instance, if
+     * activeDefragCycle needs to utilize 25% cpu, it will utilize 2.5ms, so we
+     * need to call it multiple times. */
+    long hz_ms = 1000/server.hz;
+    while (server.blocked_last_cron < server.mstime) {
+
+        /* Defrag keys gradually. */
+        if (server.active_defrag_enabled)
```

Comment:
You checked this earlier. ActiveDefragCycle() also checks if activeDefrag is enabled, so this check isn't strictly useful. It's also odd we do the check here but don't do it in serverCron.

Reply:
the first check of `active_defrag_enabled` in this function is temporary code, will be removed if / when we'll ever add more roles to this function.
but you're right, the check in databaseCron was recently removed and moved to inside ActiveDefragCycle, i'll remove this one.